### ISSUE TITLE
Extension has high CPU load

### ISSUE THREAD

**Issue by cwillinx** [shanalikhan/code-settings-sync#1393]
Type: <b>Performance Issue</b>

Open VSCode on MacOS 13.0

After the initialisation phase, a high CPU load remains, also when the IDE is idle. Having multiple instance of VSCode open multiplies the impact accordingly.

Help: Start Extension Bisect helped identify the Extension "Settings Sync" as the originator.

Extension version: 3.4.3
VS Code version: Code 1.73.0 (8fa188b2b301d36553cbc9ce1b0a146ccb93351f, 2022-11-01T15:38:50.881Z)
OS version: Darwin x64 22.1.0
Modes:
Sandboxed: No

<details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vsdfh931:30280409
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593:30376534
pythonvs932:30410667
cppdebug:30492333
vscaat:30438848
vsclangdf:30486550
c4g48928:30535728
dsvsc012:30540252
azure-dev_surveyone:30548225
vscccc:30566498
pyindex848:30577860
nodejswelcome1cf:30587006
fc301958:30595537
2e4cg342:30602488
gswce1:30605430
3d0df643:30604793
dbltrim-noruby:30604474

```

</details>

**Comment by carlosbaraza** [comment]
I had the same experience. I used "Help: Start Extension Bisect" to find the culprit. I am not sure why this extension is causing 100% CPU usage, probably blocking one CPU core and forcing the fans to get crazy.

**Comment by kelzenberg** [comment]
I can confirm this behaviour (+50% CPU usage on idle) for VSCodium through the Bisect method, too.

- Darwin Kernel Version 20.6.0
- VSCodium v1.72.2 and v1.73.1
- Sync extension v3.4.3

This seems to be the original issue #1107.

**Comment by thinkingc** [comment]
me too, uninstall it  is ok

**Comment by aprilandjan** [comment]
Yeah I came here and found that it costs terrible cpu usage. By the way, the [Extension Bisect](https://code.visualstudio.com/blogs/2021/02/16/extension-bisect) is quite handy.

**Comment by adamklepacz** [comment]
same here. Draining CPU usage over 150%

**Comment by tonsnoei** [comment]
Same here. 2 VS Code windows open. 400% CPU usage. Please fix it. Also important for our planet...

<img width="680" alt="image" src="https://user-images.githubusercontent.com/8810118/202677626-b34a1870-1621-4bbb-b062-8985d2565d58.png">

**Comment by alex-kinokon** [comment]
Already [fixed](https://github.com/shanalikhan/code-settings-sync/pull/1389) but no one has published it yet.
